packages/nuxt/src/runtime/plugins/sentry.server.ts

```diff
@@ -29,10 +66,36 @@ export default defineNitroPlugin(nitroApp => {
       captureContext: { contexts: { nuxt: structuredContext } },
       mechanism: { handled: false },
     });
+
+    await flushIfServerless();
   });
 
   // @ts-expect-error - 'render:html' is a valid hook name in the Nuxt context
   nitroApp.hooks.hook('render:html', (html: NuxtRenderHTMLContext) => {
     addSentryTracingMetaTags(html.head);
   });
 });
+
+async function flushIfServerless(): Promise<void> {
+  const isServerless = !!process.env.LAMBDA_TASK_ROOT || !!process.env.VERCEL || !!process.env.NETLIFY;
+
+  // @ts-expect-error This is not typed
+  if (GLOBAL_OBJ[Symbol.for('@vercel/request-context')]) {
+    vercelWaitUntil(flushWithTimeout());
+  } else if (isServerless) {
+    await flushWithTimeout();
+  }
+}
+
+async function flushWithTimeout(): Promise<void> {
+  const sentryClient = getClient();
+  const isDebug = sentryClient ? sentryClient.getOptions().debug : false;
```

Comment:
q: why aren't we using `DEBUG_BUILD` here?

Reply:
Because this is in the `runtime` folder of the Nuxt module and it's not possible to use it there because it's a different build context :(